https/gifts.js

```diff
@@ -16,140 +16,242 @@ const { EVENT } = require("../config/constants");
   /**
-   * marks the gift reported on the giftees account
+   * Updates the sent status of the gift
    * @inner
    * @param {object} data - details about the giftee
    * @param {string} data.user - user object or uid
-   * @param {boolean} data.value - marks it either true or false
-   * @param {string} [data.message] - message for reporting
-   * @param {object} [context] - This is used by firebase, no idea what it does, I think its added automatically
+   * @param {string} data.giftId - the uid for the gift
+   * @param {bool} data.isReporting - if the gift is being reported
    * @returns {Result}
    */
-  async ({ user, value, message }, context) => {
-    // on the giftee (current user)
-    let gifteeStatus = await markGifteeAccount(
-      { user: user },
-      { field: "reported", value: value, message: message }
-    );
-
-    let gw2Account = await markGw2Account({
-      user: user,
-      field: "reported",
-      value: value
-    });
-
-    // check result and return to frontend
-    if (gifteeStatus.success && gw2Account.success) {
-      return { success: gifteeStatus.success };
-    } else {
-      return { error: gifteeStatus.error };
+  async ({ user, giftId, isReporting }) => {
+    let gameAccountUUID = await getGameAccountUUID(user);
+    if (gameAccountUUID.error) {
+      return { error: "no API key set" };
     }
+
+    let eventDoc = db.collection(CollectionTypes.EVENTS).doc(EVENT);
+
+    let giftDoc = eventDoc
+      .collection(CollectionTypes.EVENTS__GIFTS)
+      .doc(giftId);
+
+    if (!giftDoc.exists) {
+      return { error: `Found no gifts with id: ${giftId}` };
+    }
+
+    let isGiftUpdatedSuccessfully = giftDoc
+      .set(
+        {
+          reported: isReporting ? new Date().toISOString() : null
+        },
+        { merge: true }
+      )
+      .then(() => {
+        return true;
+      })
+      .catch(() => {
+        return false;
+      });
+
+    return isGiftUpdatedSuccessfully
+      ? { success: "Successfully updated gift's reported status." }
+      : { error: "Failed updating gift's reported status." };
   }
 );
 
-module.exports = { sendGift, receiveGift, reportGift };
+// /**
+//  * @namespace sendGift
+//  * @return {sendGift~inner} - the returned function
+//  */
+// const sendGift = functions.https.onCall(
+//   /**
+//    * marks the gift sent on boith the gifter and giftees account
+//    * @inner
+//    * @param {object} data - details about the giftee
+//    * @param {string} data.user - user object or uid
+//    * @param {boolean} data.value - marks it either true or false
+//    * @param {string} data.gifteeGameAccountUUID - UUID of the giftee, if you do not know it
+//    * @param {object} [context] - This is used by firebase, no idea what it does, I think its added automatically
+//    * @returns {Result}
+//    */
+//   async ({ user, value, gifteeGameAccountUUID }, context) => {
+//     // this has to mark both the giftee and gifter
```

Comment:
Did you manually // each line?

Reply:
Haha, that would take half the afternoon. Mass comment hotkey. Will remove this later.